add 'togglePin' convenience function (#8734)
qmk/qmk_firmware: docs/internals_gpio_control.md

| `writePinLow(pin)`     | Set pin level as low, assuming it is an output   | `PORTB &= ~(1<<2)`                              | `palClearLine(pin)`                             |
| `writePin(pin, level)` | Set pin level, assuming it is an output          | `(level) ? PORTB \|= (1<<2) : PORTB &= ~(1<<2)` | `(level) ? palSetLine(pin) : palClearLine(pin)` |
| `readPin(pin)`         | Returns the level of the pin                     | `_SFR_IO8(pin >> 4) & _BV(pin & 0xF)`           | `palReadLine(pin)`                              |

> | `togglePin(pin)`       | inverts the output logic level on a pin          | `PORTB ^= (1<<2)`                               | `palToggle(pin)`                                |

Reviewer: Tweak the description so it matches the rest of the table, and use the correct function name for `palToggleLine`.
```suggestion
| `togglePin(pin)`       | Invert pin level, assuming it is an output       | `PORTB ^= (1<<2)`                               | `palToggleLine(pin)`                            |
```

> | `togglePin(pin)`       | Invert pin level, assuming it is an output       | `PORTB ^= (1<<2)`                               | `palToggleLine(pin)`                            |

Reviewer 2: I'm not a huge fan of the assumption here...

So ... what happens if the pin is set to input or high-z?

Author: should i go back to the initial documentation - which didn't use "assuming" and was talking about output pins?

Reviewer: You're always toggling output pins, regardless of the implementation.

Reviewer 3: If you call `togglePin()` on a pin currently set as input, it will still toggle the internal pullup resistor, so the pin state goes between input high and high-Z. This is just how the AVR GPIOs are designed.

Reviewer: If the behavior is the same across AVR and ARM then that's okay, but if not it should be described by it's intended interface only.

Author: ok - so worst case is that 'it does not seem to work' which would match the behavior of writePin in similar circumstances - @drashna: you okay with leaving it like this?

Reviewer 2: As long as the behavior is explicitly documented, I'm fine with it.  

It's just that if there are cases where it will not work as expected, that it's noted in documentation so somebody doesn't spend hours trying to figure out why it's not working.
